tests/PHPStan/Parser/CachedParserTest.php

```diff
@@ -0,0 +1,123 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Parser;
+
+class CachedParserTest extends \PHPUnit\Framework\TestCase
+{
+
+	/**
+	 * @dataProvider dataParseFileClearCache
+	 * @param int $cachedNodesByFileCountMax
+	 * @param int $cachedNodesByStringCountMax
+	 * @param int $cachedNodesByFileCountExpected
+	 * @param int $cachedNodesByStringCountExpected
+	 */
+	public function testParseFileClearCache(
+		int $cachedNodesByFileCountMax,
+		int $cachedNodesByStringCountMax,
+		int $cachedNodesByFileCountExpected,
+		int $cachedNodesByStringCountExpected
+	): void
+	{
+		$parser = new CachedParser(
+			$this->getParserMock(),
+			$cachedNodesByFileCountMax,
+			$cachedNodesByStringCountMax
+		);
+
+		// Add files to cache
+		for ($i = 0; $i <= $cachedNodesByFileCountMax; $i++) {
+			$parser->parseFile('file' . $i);
+		}
+
+		$reflection = new \ReflectionClass($parser);
+		$reflectionCachedNodesByFileCount = $reflection->getProperty('cachedNodesByFileCount');
+		$reflectionCachedNodesByFileCount->setAccessible(true);
```

Comment:
Never test like this. I'd rather have some public getter for this.

Reply:
Sounds reasonable. I don't wanted to add new methods.